clusterloader2/GETTING_STARTED.md

```diff
@@ -0,0 +1,227 @@
+    Params:
+      action: start
+# This label specifies for which pods we want to measure pod startup latency.
+      labelSelector: group = test-pod
+# If pod startup latency goes over threshold, test will fail.
+      threshold: 5s
+  - Identifier: WaitForControlledPodsRunning
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: start
+      apiVersion: apps/v1
+# Type of object we are waiting for
+      kind: Deployment
+      labelSelector: group = test-pod
+# How long we want to wait till object is in running state.
+      operationTimeout: 120s
+# Once we created these two measurements, we can create deployment in next step.
+- name: Create deployment
+  phases:
+# namespaceRange specifies in which namespaces we want to create this deployment.
+  - namespaceRange:
+      min: 1
+      max: 1
+# Here we an specify how many replicas of object we want to create within namespace.
+    replicasPerNamespace: 1
+    tuningSet: Uniform1qps
+    objectBundle:
+    - basename: test-deployment
+# Path to file containing template for object to be created.
+      objectTemplatePath: "deployment.yaml"
+# Parameters to fill template.
+      templateFillMap:
+        Replicas: 1
+# Once we started measurements and created deployment, now we can wait for this deployment to be running.
+# Also, we can measure pod startup latency
+- name: Wait for pods to be running
+  measurements:
+  - Identifier: WaitForControlledPodsRunning
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: gather
+  - Identifier: PodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: gather
+```
+Now, we need to prepare template for our deployment to be created. All values in brackets will be filled from template parameters:
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+# Name is parameter that will be autogenerated by ClusterLoader
+  name: {{.Name}}
+  labels:
+    group: test-pod
+spec:
+  replicas: {{.Replicas}}
+  selector:
+    matchLabels:
+      name: {{.Name}}
+  template:
+    metadata:
+      labels:
+        name: {{.Name}}
+        group: test-pod
+    spec:
+      containers:
+      - image: k8s.gcr.io/pause:3.1
+        imagePullPolicy: IfNotPresent
+        name: {{.Name}}
+        resources:
+          requests:
+            cpu: 1m
+            memory: 10M
+```
+## Execute test
+Before running test, make sure that kubeconfig current context points to kind cluster:
+```bash
+kubectl config current-context
+kind-test-cluster
+```
+To execute test, run:
+```bash
+cd clusterloader2/
+go run cmd/clusterloader.go --testconfig=config.yaml --provider=kind --kubeconfig=${HOME}/.kube/config --v=2
+```
+
+At the end of clusterloader output you should see pod startup latency:
+```json
+    {
+      "data": {
+        "Perc50": 1252.347301,
+        "Perc90": 1252.347301,
+        "Perc99": 1252.347301
+      },
+      "unit": "ms",
+      "labels": {
+        "Metric": "schedule_to_watch"
```

Comment:
Why schedule_to_watch and not "pod_startup" ?

Reply:
Fixed.